File: packages/android_intent_plus/lib/src/test_values.dart
// coverage:ignore-file
import 'package:android_intent_plus/android_intent.dart';
import 'package:flutter/foundation.dart';

@visibleForTesting
class TestValues {
  static Bundles get emptyBundle => Bundles(
        bundles: <Bundle>[
          Bundle(
            value: [],
          ),
        ],
      );

  static Bundles get bundleWithBoolean => Bundles(
        bundles: <Bundle>[
          Bundle(
            value: [
              PutBool(
                key: 'key1',
                value: true,
              ),
              PutBool(
                key: 'key2',
                value: false,
              ),
            ],
          ),
        ],
      );

  static Bundles get bundleWithBooleanArray => Bundles(
        bundles: [
          Bundle(
            value: [
              PutBoolArray(key: 'key1', value: <bool>[true, false, true]),
              PutBoolArray(key: 'key2', value: <bool>[false, true, false]),
            ],
          ),
        ],
      );

  static Bundles get bundleWithBundle => Bundles(
        bundles: [
          Bundle(
            value: [
              PutBundle(
                key: 'key1',
                value: [
                  PutString(key: 'subKey1', value: 'value1'),
                  PutInt(key: 'subKey2', value: 1),
                ],
              ),
              PutBundle(
                key: 'key2',
                value: [
                  PutString(key: 'subKey1', value: 'value2'),
                  PutInt(key: 'subKey2', value: 2),
                ],
              ),
            ],
          ),
        ],
      );

  static Bundles get bundleWithByte => Bundles(
        bundles: [
          Bundle(
            value: [
              PutByte(key: 'key1', value: -128),
              PutByte(key: 'key2', value: 127)
            ],
          ),
        ],
      );

  static Bundles get bundleWithByteInvalidLowerBound => Bundles(
        bundles: [
          Bundle(
            value: [
              PutByte(key: 'key1', value: -129),
              PutByte(key: 'key2', value: 127)
            ],
          ),
        ],
      );

  static Bundles get bundleWithByteInvalidUpperBound => Bundles(
        bundles: [
          Bundle(
            value: [
              PutByte(key: 'key1', value: -128),
              PutByte(key: 'key2', value: 128)
            ],
          ),
        ],
      );

  static Bundles get bundleWithByteArray => Bundles(
        bundles: [
          Bundle(
            value: [
              PutByteArray(key: 'key1', value: [-128, 0, 127]),
              PutByteArray(key: 'key2', value: [127, 0, -128])
            ],
          ),
        ],
      );

  static Bundles get bundleWithChar => Bundles(
        bundles: [
          Bundle(
            value: [
              PutChar(key: 'key1', value: 'a'),
              PutChar(key: 'key2', value: 'b'),
            ],
          ),
        ],
      );

  static Bundles get bundleWithCharInvalidChar => Bundles(
        bundles: [
          Bundle(
            value: [
              PutChar(key: 'key1', value: 'value1'),
              PutChar(key: 'key2', value: 'value2'),
            ],
          ),
        ],
      );

  static Bundles get bundleWithCharArray => Bundles(
        bundles: [
          Bundle(
            value: [
              PutCharArray(key: 'key1', value: ['a', 'b', 'c']),
              PutCharArray(key: 'key2', value: ['c', 'b', 'a']),
            ],
          ),
        ],
      );

  static Bundles get bundleWithCharArrayInvalidChar => Bundles(
        bundles: [
          Bundle(
            value: [
              PutCharArray(key: 'key1', value: ['a', 'b', 'c']),
              PutCharArray(key: 'key2', value: ['c', 'b', 'abc']),
            ],
          ),
        ],
      );

  static Bundles get bundleWithCharSequence => Bundles(
        bundles: [
          Bundle(
            value: [
              PutCharSequence(
                key: 'key1',
                value: 'value1',
              ),
              PutCharSequence(
                key: 'key2',
                value: 'value2',
              ),
            ],
          ),
        ],
      );

  static Bundles get bundleWithCharSequenceArray => Bundles(
        bundles: [
          Bundle(
            value: [
              PutCharSequenceArray(
                key: 'key1',
                value: ['value1', 'value2', 'value3'],
              ),
              PutCharSequenceArray(
                key: 'key2',
                value: ['value4', 'value5', 'value6'],
              ),
            ],
          ),
        ],
      );

  static Bundles get bundleWithCharSequenceArrayList => Bundles(
        bundles: [
          Bundle(
            value: [
              PutCharSequenceArrayList(
                key: 'key1',
                value: ['value1', 'value2', 'value3'],
              ),
              PutCharSequenceArrayList(
                key: 'key2',
                value: ['value4', 'value5', 'value6'],
              ),
            ],
          ),
        ],
      );

  static Bundles get bundleWithByteArrayInvalidLowerBound => Bundles(
        bundles: [
          Bundle(
            value: [
              PutByteArray(key: 'key1', value: [-129, 0, 127]),
              PutByteArray(key: 'key2', value: [127, 0 - 128])
            ],
          ),
        ],
      );

  static Bundles get bundleWithByteArrayInvalidUpperBound => Bundles(
        bundles: [
          Bundle(
            value: [
              PutByteArray(key: 'key1', value: [-128, 0, 128]),
              PutByteArray(key: 'key2', value: [127, 0 - 128])
            ],
          ),
        ],
      );

  static Bundles get bundleWithDouble => Bundles(
        bundles: [
          Bundle(
            value: [
              PutDouble(key: 'key1', value: -9.999999e96),
              PutDouble(key: 'key2', value: 1e-101),
              PutDouble(key: 'key3', value: 9.999999e96),
            ],
          ),
        ],
      );

  static Bundles get bundleWithDoubleArray => Bundles(
        bundles: [
          Bundle(
            value: [
              PutDoubleArray(
                key: 'key1',
                value: [-9.999999e96, 1e-101, 9.999999e96],
              ),
              PutDoubleArray(
                key: 'key2',
                value: [9.999999e96, 1e-101, -9.999999e96],
              ),
            ],
          ),
        ],
      );

  static Bundles get bundleWithFloat => Bundles(
        bundles: [
          Bundle(
            value: [
              PutFloat(key: 'key1', value: -3.4028235e+38),
              PutFloat(key: 'key2', value: 1.4e-45),
              PutFloat(key: 'key3', value: 3.4028235e+38),
            ],
          ),
        ],
      );

  static Bundles get bundleWithFloatArray => Bundles(
        bundles: [
          Bundle(
            value: [
              PutFloatArray(
                key: 'key1',
                value: [-3.4028235e+38, 1.4e-45, 3.4028235e+38],
              ),
              PutFloatArray(
                key: 'key2',
                value: [3.4028235e+38, 1.4e-45, -3.4028235e+38],
              ),
            ],
          ),
        ],
      );

  static Bundles get bundleWithInt => Bundles(
        bundles: [
          Bundle(
            value: [
              PutInt(key: 'key1', value: -2147483648),
              PutInt(key: 'key2', value: 2147483647)
            ],
          ),
        ],
      );

  static Bundles get bundleWithIntInvalidLowerBound => Bundles(
        bundles: [
          Bundle(
            value: [
              PutInt(key: 'key1', value: -2147483649),
              PutInt(key: 'key2', value: 2147483647)
            ],
          ),
        ],
      );

  static Bundles get bundleWithIntInvalidUpperBound => Bundles(
        bundles: [
          Bundle(
            value: [
              PutInt(key: 'key1', value: -2147483648),
              PutInt(key: 'key2', value: 2147483648)
            ],
          ),
        ],
      );

  static Bundles get bundleWithIntArray => Bundles(
        bundles: [
          Bundle(
            value: [
              PutIntArray(key: 'key1', value: [-2147483648, 0, 2147483647]),
              PutIntArray(key: 'key2', value: [2147483647, 0, -2147483648]),
            ],
          ),
        ],
      );

  static Bundles get bundleWithIntArrayInvalidLowerBound => Bundles(
        bundles: [
          Bundle(
            value: [
              PutIntArray(key: 'key1', value: [-2147483649, 0, 2147483647]),
              PutIntArray(key: 'key2', value: [2147483647, 0, -2147483648]),
            ],
          ),
        ],
      );

  static Bundles get bundleWithIntArrayInvalidUpperBound => Bundles(
        bundles: [
          Bundle(
            value: [
              PutIntArray(key: 'key1', value: [-2147483648, 0, 2147483648]),
              PutIntArray(key: 'key2', value: [2147483647, 0, -2147483648]),
            ],
          ),
        ],
      );

  static Bundles get bundleWithIntegerArrayList => Bundles(
        bundles: [
          Bundle(
            value: [
              PutIntegerArrayList(
                  key: 'key1', value: [-2147483648, 0, 2147483647]),
              PutIntegerArrayList(
                  key: 'key2', value: [2147483647, 0, -2147483648]),
            ],
          ),
        ],
      );

  static Bundles get bundleWithIntegerArrayListInvalidLowerBound => Bundles(
        bundles: [
          Bundle(
            value: [
              PutIntegerArrayList(
                  key: 'key1', value: [-2147483649, 0, 2147483647]),
              PutIntegerArrayList(
                  key: 'key2', value: [2147483647, 0, -2147483648]),
            ],
          ),
        ],
      );

  static Bundles get bundleWithIntegerArrayListInvalidUpperBound => Bundles(
        bundles: [
          Bundle(
            value: [
              PutIntegerArrayList(
                  key: 'key1', value: [-2147483648, 0, 2147483648]),
              PutIntegerArrayList(
                  key: 'key2', value: [2147483647, 0, -2147483648]),
            ],
          ),
        ],
      );

  static Bundles get bundleWithLong => Bundles(
        bundles: [
          Bundle(
            value: [
              PutLong(key: 'key1', value: -9223372036854775808),
              PutLong(key: 'key2', value: 9223372036854775807)
            ],
          ),
        ],
      );

  static Bundles get bundleWithLongArray => Bundles(
        bundles: [
          Bundle(
            value: [
              PutLongArray(
                key: 'key1',
                value: [-9223372036854775808, 0, 9223372036854775807],
              ),
              PutLongArray(
                key: 'key2',
                value: [9223372036854775807, 0, -9223372036854775808],
              ),
            ],
          ),
        ],
      );

  static Bundles get bundleWithParcelable => Bundles(
        bundles: [
          Bundle(
            value: [
              PutParcelable(
                key: 'key1',
                value: Bundle(value: [
                  PutString(key: 'subKey1', value: 'value1'),
                  PutInt(key: 'subKey2', value: 1),
                ]),
              ),
              PutParcelable(
                key: 'key2',
                value: Bundle(value: [
                  PutString(key: 'subKey1', value: 'value2'),
                  PutInt(key: 'subKey2', value: 2),
                ]),
              ),
            ],
          ),
        ],
      );

  static Bundles get bundleWithParcelableArray => Bundles(
        bundles: [
          Bundle(
            value: [
              PutParcelableArray(
                key: 'key1',
                value: [
                  Bundle(value: [
                    PutString(key: 'subKey1', value: 'value1'),
                    PutInt(key: 'subKey2', value: 1),
                  ]),
                  Bundle(value: [
                    PutString(key: 'subKey1', value: 'value2'),
                    PutInt(key: 'subKey2', value: 2),
                  ])
                ],
              ),
              PutParcelableArray(
                key: 'key2',
                value: [
                  Bundle(value: [
                    PutString(key: 'subKey1', value: 'value3'),
                    PutInt(key: 'subKey2', value: 3),
                  ]),
                  Bundle(value: [
                    PutString(key: 'subKey1', value: 'value4'),
                    PutInt(key: 'subKey2', value: 4),
                  ])
                ],
              ),
            ],
          ),
        ],
      );

  static Bundles get bundleWithParcelableArrayList => Bundles(
        bundles: [
          Bundle(
            value: [
              PutParcelableArrayList(
                key: 'key1',
                value: [
                  Bundle(value: [
                    PutString(key: 'subKey1', value: 'value1'),
                    PutInt(key: 'subKey2', value: 1),
                  ]),
                  Bundle(value: [
                    PutString(key: 'subKey1', value: 'value2'),
                    PutInt(key: 'subKey2', value: 2),
                  ])
                ],
              ),
              PutParcelableArrayList(
                key: 'key2',
                value: [
                  Bundle(value: [
                    PutString(key: 'subKey1', value: 'value3'),
                    PutInt(key: 'subKey2', value: 3),
                  ]),
                  Bundle(value: [
                    PutString(key: 'subKey1', value: 'value4'),
                    PutInt(key: 'subKey2', value: 4),
                  ])
                ],
              ),
            ],
          ),
        ],
      );

  static Bundles get bundleWithShort => Bundles(
        bundles: [
          Bundle(
            value: [
              PutShort(key: 'key1', value: -32768),
              PutShort(key: 'key2', value: 32767)
            ],
          ),
        ],
      );

  static Bundles get bundleWithShortInvalidLowerBound => Bundles(
        bundles: [
          Bundle(
            value: [
              PutShort(key: 'key1', value: -32769),
              PutShort(key: 'key2', value: 32767)
            ],
          ),
        ],
      );

  static Bundles get bundleWithShortInvalidUpperBound => Bundles(
        bundles: [
          Bundle(
            value: [
              PutShort(key: 'key1', value: -32768),
              PutShort(key: 'key2', value: 32768)
            ],
          ),
        ],
      );

  static Bundles get bundleWithShortArray => Bundles(
        bundles: [
          Bundle(
            value: [
              PutShortArray(key: 'key1', value: [-32768, 0, 32767]),
              PutShortArray(key: 'key2', value: [32767, 0, -32768]),
            ],
          ),
        ],
      );

  static Bundles get bundleWithShortArrayInvalidLowerBound => Bundles(
        bundles: [
          Bundle(
            value: [
              PutShortArray(key: 'key1', value: [-32769, 0, 32767]),
              PutShortArray(key: 'key2', value: [32767, 0, -32768]),
            ],
          ),
        ],
      );

  static Bundles get bundleWithShortArrayInvalidUpperBound => Bundles(
        bundles: [
          Bundle(
            value: [
              PutShortArray(key: 'key1', value: [-32768, 0, 32768]),
              PutShortArray(key: 'key2', value: [32767, 0, -32768]),
            ],
          ),
        ],
      );

  static Bundles get bundleWithString => Bundles(
        bundles: [
          Bundle(
            value: [
              PutString(key: 'key1', value: "value1"),
              PutString(key: 'key2', value: "value2")
            ],
          ),
        ],
      );

  static Bundles get bundleWithStringArray => Bundles(
        bundles: [
          Bundle(
            value: [
              PutStringArray(
                  key: 'key1', value: ['value1', 'value2', 'value3']),
              PutStringArray(key: 'key2', value: ['value4', 'value5', 'value6'])
            ],
          ),
        ],
      );

  static Bundles get bundleWithStringArrayList => Bundles(
        bundles: [
          Bundle(
            value: [
              PutStringArrayList(
                  key: 'key1', value: ['value1', 'value2', 'value3']),
              PutStringArrayList(
                  key: 'key2', value: ['value4', 'value5', 'value6'])
            ],
          ),
        ],
      );

  static String get bundleWithUnknownPutBaseInBundle => '''
{
  "javaClass": "Bundles",
  "value": [
    {
      "javaClass": "Bundle",
      "value": [
        {
          "key": "key1",
          "javaClass": "PutStringUNKNOWN",
          "value": "value1"
        }
      ]
    }
  ]
}''';

  static String get bundleWithUnknownBundleInParcelable => '''
{
  "javaClass": "Bundles",
  "value": [
    {
      "javaClass": "Bundle",
      "value": [
        {
          "key": "key1",
          "javaClass": "PutParcelable",
          "value": {
            "javaClass": "BundleUNKNOWN",
            "value": []
          }
        }
      ]
    }
  ]
}
''';

  static String get bundleWithUnknownBundleInParcelableArray => '''
{
  "javaClass": "Bundles",
  "value": [
    {
      "javaClass": "Bundle",
      "value": [
        {
          "key": "key1",
          "javaClass": "PutParcelableArray",
          "value": [
            {
              "javaClass": "BundleUNKNOWN",
              "value": []
            }
          ]
        }
      ]
    }
  ]
}
''';

  static String get bundleWithUnknownBundleInParcelableArrayList => '''
{
  "javaClass": "Bundles",
  "value": [
    {
      "javaClass": "Bundle",
      "value": [
        {
          "key": "key1",
          "javaClass": "PutParcelableArrayList",
          "value": [
            {
              "javaClass": "BundleUNKNOWN",
              "value": []
            }
          ]
        }
      ]
    }
  ]
}
''';
}

File: packages/android_intent_plus/android/src/test/java/dev/fluttercommunity/plus/androidintent/TestValues.java
package dev.fluttercommunity.plus.androidintent;

import dev.fluttercommunity.plus.androidintent.Bundle.Bundles;
import dev.fluttercommunity.plus.androidintent.Bundle.ParcelableClasses.Bundle;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutBool;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutBoolArray;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutBundle;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutByte;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutByteArray;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutChar;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutCharArray;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutCharSequence;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutCharSequenceArray;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutCharSequenceArrayList;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutDouble;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutDoubleArray;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutFloat;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutFloatArray;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutInt;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutIntArray;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutIntegerArrayList;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutLong;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutLongArray;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutParcelable;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutParcelableArray;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutParcelableArrayList;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutShort;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutShortArray;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutString;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutStringArray;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.PutStringArrayList;
import java.util.Arrays;
import java.util.Collections;

@SuppressWarnings("ArraysAsListWithZeroOrOneArgument")
public class TestValues {
  static Bundles emptyBundle() {
    return new Bundles(Collections.singletonList(new Bundle()));
  }

  public static Bundles bundleWithBoolean() {
    return new Bundles(
        Arrays.asList(
            new Bundle(Arrays.asList(new PutBool("key1", true), new PutBool("key2", false)))));
  }

  public static Bundles bundleWithBooleanArray() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutBoolArray("key1", Arrays.asList(true, false, true)),
                    new PutBoolArray("key2", Arrays.asList(false, true, false))))));
  }

  public static Bundles bundleWithBundle() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutBundle(
                        "key1",
                        Arrays.asList(
                            new PutString("subKey1", "value1"), new PutInt("subKey2", 1))),
                    new PutBundle(
                        "key2",
                        Arrays.asList(
                            new PutString("subKey1", "value2"), new PutInt("subKey2", 2)))))));
  }

  public static Bundles bundleWithByte() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(new PutByte("key1", (byte) -128), new PutByte("key2", (byte) 127)))));
  }

  public static Bundles bundleWithByteArray() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutByteArray("key1", Arrays.asList((byte) -128, (byte) 0, (byte) 127)),
                    new PutByteArray("key2", Arrays.asList((byte) 127, (byte) 0, (byte) -128))))));
  }

  public static Bundles bundleWithChar() {
    return new Bundles(
        Arrays.asList(
            new Bundle(Arrays.asList(new PutChar("key1", 'a'), new PutChar("key2", 'b')))));
  }

  public static Bundles bundleWithCharArray() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutCharArray("key1", Arrays.asList('a', 'b', 'c')),
                    new PutCharArray("key2", Arrays.asList('c', 'b', 'a'))))));
  }

  public static Bundles bundleWithCharSequence() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutCharSequence("key1", "value1"),
                    new PutCharSequence("key2", "value2")))));
  }

  public static Bundles bundleWithCharSequenceArray() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutCharSequenceArray("key1", Arrays.asList("value1", "value2", "value3")),
                    new PutCharSequenceArray(
                        "key2", Arrays.asList("value4", "value5", "value6"))))));
  }

  public static Bundles bundleWithCharSequenceArrayList() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutCharSequenceArrayList(
                        "key1", Arrays.asList("value1", "value2", "value3")),
                    new PutCharSequenceArrayList(
                        "key2", Arrays.asList("value4", "value5", "value6"))))));
  }

  public static Bundles bundleWithDouble() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutDouble("key1", -9.999999e+96),
                    new PutDouble("key2", 1e-101),
                    new PutDouble("key3", 9.999999e+96)))));
  }

  public static Bundles bundleWithDoubleArray() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutDoubleArray("key1", Arrays.asList(-9.999999e+96, 1e-101, 9.999999e+96)),
                    new PutDoubleArray(
                        "key2", Arrays.asList(9.999999e+96, 1e-101, -9.999999e+96))))));
  }

  public static Bundles bundleWithFloat() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutFloat("key1", -3.4028235e+38f),
                    new PutFloat("key2", 1.4e-45f),
                    new PutFloat("key3", 3.4028235e+38f)))));
  }

  public static Bundles bundleWithFloatArray() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutFloatArray(
                        "key1", Arrays.asList(-3.4028235e+38f, 1.4e-45f, 3.4028235e+38f)),
                    new PutFloatArray(
                        "key2", Arrays.asList(3.4028235e+38f, 1.4e-45f, -3.4028235e+38f))))));
  }

  public static Bundles bundleWithInt() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(new PutInt("key1", -2147483648), new PutInt("key2", 2147483647)))));
  }

  public static Bundles bundleWithIntArray() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutIntArray("key1", Arrays.asList(-2147483648, 0, 2147483647)),
                    new PutIntArray("key2", Arrays.asList(2147483647, 0, -2147483648))))));
  }

  public static Bundles bundleWithIntegerArrayList() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutIntegerArrayList("key1", Arrays.asList(-2147483648, 0, 2147483647)),
                    new PutIntegerArrayList("key2", Arrays.asList(2147483647, 0, -2147483648))))));
  }

  public static Bundles bundleWithLong() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutLong("key1", -9223372036854775808L),
                    new PutLong("key2", 9223372036854775807L)))));
  }

  public static Bundles bundleWithLongArray() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutLongArray(
                        "key1", Arrays.asList(-9223372036854775808L, 0L, 9223372036854775807L)),
                    new PutLongArray(
                        "key2", Arrays.asList(9223372036854775807L, 0L, -9223372036854775808L))))));
  }

  public static Bundles bundleWithShort() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutShort("key1", (short) -32768), new PutShort("key2", (short) 32767)))));
  }

  public static Bundles bundleWithShortArray() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutShortArray(
                        "key1", Arrays.asList((short) -32768, (short) 0, (short) 32767)),
                    new PutShortArray(
                        "key2", Arrays.asList((short) 32767, (short) 0, (short) -32768))))));
  }

  public static Bundles bundleWithString() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(new PutString("key1", "value1"), new PutString("key2", "value2")))));
  }

  public static Bundles bundleWithStringArray() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutStringArray("key1", Arrays.asList("value1", "value2", "value3")),
                    new PutStringArray("key2", Arrays.asList("value4", "value5", "value6"))))));
  }

  public static Bundles bundleWithStringArrayList() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutStringArrayList("key1", Arrays.asList("value1", "value2", "value3")),
                    new PutStringArrayList("key2", Arrays.asList("value4", "value5", "value6"))))));
  }

  public static Bundles bundleWithParcelable() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutParcelable(
                        "key1",
                        new Bundle(
                            Arrays.asList(
                                new PutString("subKey1", "value1"), new PutInt("subKey2", 1)))),
                    new PutParcelable(
                        "key2",
                        new Bundle(
                            Arrays.asList(
                                new PutString("subKey1", "value2"), new PutInt("subKey2", 2))))))));
  }

  public static Bundles bundleWithParcelableArray() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutParcelableArray(
                        "key1",
                        Arrays.asList(
                            new Bundle(
                                Arrays.asList(
                                    new PutString("subKey1", "value1"), new PutInt("subKey2", 1))),
                            new Bundle(
                                Arrays.asList(
                                    new PutString("subKey1", "value2"),
                                    new PutInt("subKey2", 2))))),
                    new PutParcelableArray(
                        "key2",
                        Arrays.asList(
                            new Bundle(
                                Arrays.asList(
                                    new PutString("subKey1", "value3"), new PutInt("subKey2", 3))),
                            new Bundle(
                                Arrays.asList(
                                    new PutString("subKey1", "value4"),
                                    new PutInt("subKey2", 4)))))))));
  }

  public static Bundles bundleWithParcelableArrayList() {
    return new Bundles(
        Arrays.asList(
            new Bundle(
                Arrays.asList(
                    new PutParcelableArrayList(
                        "key1",
                        Arrays.asList(
                            new Bundle(
                                Arrays.asList(
                                    new PutString("subKey1", "value1"), new PutInt("subKey2", 1))),
                            new Bundle(
                                Arrays.asList(
                                    new PutString("subKey1", "value2"),
                                    new PutInt("subKey2", 2))))),
                    new PutParcelableArrayList(
                        "key2",
                        Arrays.asList(
                            new Bundle(
                                Arrays.asList(
                                    new PutString("subKey1", "value3"), new PutInt("subKey2", 3))),
                            new Bundle(
                                Arrays.asList(
                                    new PutString("subKey1", "value4"),
                                    new PutInt("subKey2", 4)))))))));
  }
}
